### PR TITLE
fix sorting tolerations in case the keys are equal

### LINKED ISSUE #61250: Duplicated toleration output in kubectl describe
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Trigger the following issue when testing https://github.com/kubernetes/kubernetes/pull/61161 

```
[root@ib15b13 kubernetes]# kc describe pods fluentd-elasticsearch-n75xl | grep Tolerations -A 8
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
                 node.kubernetes.io/unschedulable:NoSchedule
                 node.kubernetes.io/unschedulable:NoSchedule
                 node.kubernetes.io/unschedulable:NoSchedule
                 node.kubernetes.io/unschedulable:NoSchedule
```

```
[root@ib15b13 kubernetes]# kc get pods fluentd-elasticsearch-n75xl -o yaml | grep tolerations -A 20
  tolerations:
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
```

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
[root@ib15b13 kubernetes]# kc version
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-alpha.0.1197+bae1f72cc2ac65-dirty", GitCommit:"bae1f72cc2ac65af8273f3808238c9c5ce671a2c", GitTreeState:"dirty", BuildDate:"2018-03-16T00:30:28Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-alpha.0.1197+bae1f72cc2ac65-dirty", GitCommit:"bae1f72cc2ac65af8273f3808238c9c5ce671a2c", GitTreeState:"dirty", BuildDate:"2018-03-16T00:30:28Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
```
[root@ib15b13 kubernetes]# cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.2 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.2 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.2"
```
- Kernel (e.g. `uname -a`): `Linux ib15b13.eng.platformlab.ibm.com 3.10.0-693.21.1.el7.x86_64 #1 SMP Fri Feb 23 18:54:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

/sig cli

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
/kind bug
/sig cli

find a better way to sort `api.Toleration`, in case the key are the same

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61250

**Special notes for your reviewer**:
/cc @k82cn @kubernetes/sig-cli-api-reviews 
**Release note**:

```release-note
fix sorting tolerations in case the keys are equal
```

### CODE REVIEW

File: pkg/printers/internalversion/describe.go
Comment (soltysh) [inline pkg/printers/internalversion/describe.go]: Why not using `sets.String` which for one will ensure you have exactly one key and then iterate over its `List` which returns sorted keys?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/printers/internalversion/describe.go
Comment (dixudx) [reply pkg/printers/internalversion/describe.go]: `tolerations` is a list, where we cannot guarantee the key is unique. Just like the example in #61250 shows.

If we are using `sets.String`, we will ignore the `toleration` with the same key.

Comment (soltysh) [reply pkg/printers/internalversion/describe.go]: What I was suggesting, was to use the `sets.String` to sort the keys like before and get rid of duplicates. But looking once again at your solution I think your approach is much better, thanks!

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by dixudx** [comment]
ping @smarterclayton for approval. Thanks.

**Comment by soltysh** [comment]
@dixudx ensure tests are green before getting approval, pls

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**
